api/pkg/transformer/server/grpc/predictor.go

```diff
@@ -0,0 +1,158 @@
+package grpc
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/afex/hystrix-go/hystrix"
+	hystrixGo "github.com/afex/hystrix-go/hystrix"
+	hystrixpkg "github.com/caraml-dev/merlin/pkg/hystrix"
+	"github.com/caraml-dev/merlin/pkg/transformer/server/config"
+	upiv1 "github.com/caraml-dev/universal-prediction-interface/gen/go/grpc/caraml/upi/v1"
+	"github.com/go-coldbrew/grpcpool"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/keepalive"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/protobuf/encoding/protojson"
+)
+
+type upiPredictorClient interface {
+	predict(ctx context.Context, payload *upiv1.PredictValuesRequest) (*upiv1.PredictValuesResponse, error)
+	close() error
+}
+
+func newUPIPredictorClient(opts *config.Options) (upiPredictorClient, error) {
+	if opts.PredictorUPIHTTPEnabled {
```

Comment:
may need to check if transformer enabled as well here?

Reply:
it should be checked before deployment